### PR TITLE
Add atomic descriptors from Mendeleev

### PR DESCRIPTION
**Pull request by pzhanggit**
Add atomic descriptors from Mendeleev (https://mendeleev.readthedocs.io/en/stable/index.html)

### CODE REVIEW

File: tests/test_atomicdescriptors.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import subprocess
+
+
+@pytest.mark.mpi_skip()
+def pytest_atomicdescriptors():
+    file_path = os.path.join(
+        os.path.dirname(__file__), "..", "hydragnn/utils/atomicdescriptors.py"
+    )
+    return_code = subprocess.call(["python", file_path])
```
Comment (allaffa) [inline tests/test_atomicdescriptors.py:23]: Very nice! I did not know about this.

File: hydragnn/utils/atomicdescriptors.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        group_id = F.one_hot(torch.Tensor(group_id).long(), num_classes=-1)
+        return group_id
+
+    def get_group_id(self, element):
+        if mendeleev.element(element).group_id is not None:
+            return mendeleev.element(element).group_id
+        else:  # 59-71
+            print("Warning: manually assign group_id for element: %s" % element)
+            if mendeleev.element(element).atomic_number < 71:
+                return mendeleev.element(element).atomic_number - 57 + 3
```
Comment (allaffa) [inline hydragnn/utils/atomicdescriptors.py:82]: Can you explain why you have to do this?

File: hydragnn/utils/atomicdescriptors.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        ie = F.one_hot(torch.Tensor(ie).long(), num_classes=10)
+        return ie
+
+    def get_atom_features(self, atomtype):
+        if isinstance(atomtype, str):
+            atomtype = mendeleev.element(atomtype).atomic_number
+        return torch.tensor(self.atom_embeddings[str(atomtype)])
+
+
+if __name__ == "__main__":
```
Comment (allaffa) [inline hydragnn/utils/atomicdescriptors.py:198]: Do we need this call? I think you used this to make sure that the code developed was working fine. But I don't think this main function should stay. Do you agree?

Comment (allaffa) [inline hydragnn/utils/atomicdescriptors.py:198]: I think the code would be better structures if this function were to be moved inside `test_atomicdescriptors.py`
Do you agree?

**Review by allaffa — COMMENTED** [review]
First round of review

File: hydragnn/utils/atomicdescriptors.py (hunk 2 above)
Comment (pzhanggit) [reply hydragnn/utils/atomicdescriptors.py:82]: `mendeleev` library returns `None` for those elements.

File: hydragnn/utils/atomicdescriptors.py (hunk 3 above)
Comment (pzhanggit) [reply hydragnn/utils/atomicdescriptors.py:198]: Both would work. I prefer to have it here for convenient/easy testing/debugging. What do you think?

File: hydragnn/utils/atomicdescriptors.py (hunk 2 above)
Comment (allaffa) [reply hydragnn/utils/atomicdescriptors.py:82]: Just throw an error if Mendeleev does not support information for a specific element, regardless of whether it is natural or synthetic.

**Review by allaffa — CHANGES_REQUESTED** [review]

**Review by allaffa — APPROVED** [review]

**Pull request merged**